ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java

```diff
@@ -267,6 +267,12 @@ private void produceQueue() {
 
                 if (syncDone && (importResult == IMPORTED_BEST || importResult == IMPORTED_NOT_BEST)) {
                     if (logger.isDebugEnabled()) logger.debug("Block dump: " + Hex.toHexString(wrapper.getBlock().getEncoded()));
+
+                    // Propagate block to the net after successful import
+                    if (wrapper.isNewBlock()) {
+                        Channel receivedFrom = channelManager.getActivePeer(wrapper.getNodeId());
+                        channelManager.sendNewBlock(wrapper.getBlock(), receivedFrom);
+                    }
```

Comment:
Can you please submit this logic to a separate async executor? 
This code doesn't seem to do any blocking stuff in the current implementation, but I'd prefer to invoke any network related stuff asynchronously from this method just for the case.